### DISCUSSION TITLE
How to convert lon/lat offset to meter offset?

### DISCUSSION THREAD

**Discussion by gallickgunner** [visgl/deck.gl#5544] [Q&A] [answered]
Hi, It seems the function `lngLatDeltaToMeterOffset` was removed from `WebMercatorViewPort()` to convert between the 2 systems.  Were they moved somewhere else or what? I don't find any information on that in the Upgrade guide, just the fact that they got removed. 

Thanks

**Comment by Pessimistress** [comment, marked as answer]
The method was dropped in v8.0.0: https://deck.gl/docs/upgrade-guide#removed

You can use [@turf.js/distance](http://turfjs.org/docs/#distance) to get more accurate conversion.

May I ask why you were using it?

**Reply by gallickgunner** [reply]
Sure, we implemented a custom brushing thingy where all the arcs get faded out (not completely) based on if they are inside/outside a certain radius around a specific/picked arc. 

So during the calculations, we had the offsets of all the arcs from that picked arc, converted that to Meter offsets, used the distance formula to get distance and checked whether it was more/less than that certain radius. There might be a better way though, haven't thought about it. If you have one in mind, I'm all ears :)

That turf.js distance basically envelopes converting the lngLat offsets to meter offsets and then finding the distance if I got it right. That's basically what I needed anyways. Thanks

**Reply by Pessimistress** [reply]
turf.js is the way to go. The old viewport method was only an approximation with acceptable error if your two points were within 5km or so from the viewport center.
